### PR TITLE
Refactor awsElasticBlockStore to seperate builder and cleaner

### PR DESCRIPTION
**Pull request by jiangyaoguo**
This is a try to #9139. Only change  "awsElasticBlockStore'  for discuss.

### CODE REVIEW

**Comment by jiangyaoguo** [comment]
@markturansky @pmorie

**Comment by markturansky** [comment]
Looks pretty good at first glance.  You should include unit tests that verify the new type assertions.

**Comment by pmorie** [comment]
ok to test

**Comment by pmorie** [comment]
@jiangyaoguo Does this change affect and E2E tests?

**Comment by jiangyaoguo** [comment]
@pmorie  It doesn't affect e2e tests. All the changes are inside.

File: pkg/volume/aws_ebs/aws_ebs.go [hunk 1]
```diff
@@ -109,14 +109,14 @@ func (plugin *awsElasticBlockStorePlugin) NewCleaner(volName string, podUID type
...
 
 func (plugin *awsElasticBlockStorePlugin) newCleanerInternal(volName string, podUID types.UID, manager pdManager, mounter mount.Interface) (volume.Cleaner, error) {
-	return &awsElasticBlockStore{
+	return &awsElasticBlockStoreCleaner{&awsElasticBlockStore{
 		podUID:      podUID,
 		volName:     volName,
 		manager:     manager,
 		mounter:     mounter,
 		diskMounter: &awsSafeFormatAndMount{mounter, exec.New()},
 		plugin:      plugin,
```
Comment (markturansky) [inline pkg/volume/aws_ebs/aws_ebs.go]: I think we can remove unused attributes from the Cleaner and Builder.   For example, a Cleaner will never call diskMounter.  You would need to look at the functionality of the Cleaner to know which attributes can be removed from the cleaner struct.

**Comment by jiangyaoguo** [comment]
@markturansky If this change make sense, I'll spread it to other plugins.

**Comment by pmorie** [comment]
@yaoguo I will probably do this refactor for emptyDir, gitRepo, and secrets

**Comment by pmorie** [comment]
@jiangyaoguo I think that `awsElasticBlockStore` should have only the fields that are passed to it via `newCleanerInternal`, and the rest should move down to the builder.

**Comment by pmorie** [comment]
@jiangyaoguo I would also add type assertions around the new builder / cleaner types:

```go
var _ volume.Builder = &awsElasticBlockStoreBuilder{}
```

**Comment by jiangyaoguo** [comment]
@thockin

File: pkg/volume/aws_ebs/aws_ebs.go [hunk 2]
```diff
@@ -45,6 +45,8 @@ type awsElasticBlockStorePlugin struct {
 }
 
 var _ volume.VolumePlugin = &awsElasticBlockStorePlugin{}
+var _ volume.Builder = &awsElasticBlockStoreBuilder{}
```
Comment (pmorie) [inline pkg/volume/aws_ebs/aws_ebs.go]: I would relocate these to be near the definition of the types.

**Comment by mikedanese** [comment]
Still merge conflict.

**Pull request merged**